### ISSUE TITLE
`pyversion` (in `doctest`) does not work as expected with the `==` operand

### ISSUE THREAD

**Issue by dhermes** [sphinx-doc/sphinx#4183]
### Problem

Using `:pyversion: == 3.6` (or any other version) will **always** result in the test being skipped.

This is because `LooseVersion` does not work the same way `sphinx.ext.doctest.compare_version` expects it to

```python
>>> from distutils.version import LooseVersion
>>> py36 = LooseVersion('3.6')
>>> py363 = LooseVersion('3.6.3')
>>> py36 == py363
False
```

#### Procedure to reproduce the problem

When specifying

```
.. doctest:: cheez
   :pyversion: == 3.6

   >>> 10 + 4
   14

.. doctest:: burger
   :pyversion: >= 3.6

   >>> import sys
   >>> sys.version
   '3.6.3 (default, Oct 10 2017, 10:20:57) \n[GCC 5.4.0 20160609]'

.. doctest:: burger-too
   :pyversion: == 3.6.3

   >>> 1.0 / 32
   0.03125
```

and running:

```
sphinx-build -W -b doctest -d build/doctrees . build/doctest
```

the `cheez` doctest is not run:

```
Document: index
---------------
1 items had no tests:
    cheez
2 items passed all tests:
   2 tests in burger
   1 tests in burger-too
3 tests in 3 items.
3 passed and 0 failed.
Test passed.

Doctest summary
===============
    3 tests
    0 failures in tests
    0 failures in setup code
    0 failures in cleanup code
```

#### Expected results

Run the code

### Reproducible project / your project

See above

### Environment info
- OS: Confirmed on Linux and OS X
- Python version: 3.6
- Sphinx version: 1.6.4

**Comment by tk0miya** [comment]
I just made a new PR #4210 to resolve this.
Could you check this and give me your opinion please?

**Comment by dhermes** [comment]
@tk0miya That doesn't quite resolve this issue. The problem is in `sphinx.ext.doctest.compare_version()`.

I realize saying `>= 3.6, < 3.7` will be **equivalent** to `== 3.6`, but it will still leave `==` "broken" / with unexpected behavior.

**Comment by tk0miya** [comment]
I think the behavior is a specification of `:pyversion:` option from its beginning. I don't know which is natural, but we have to take care of compatibility. There are some documents which already use it.

**Comment by dhermes** [comment]
I'm not sure how you mean "is a specification". I understand the desire to stay compatible, but right now `== 3.6` will not match **any** version at all. So anyone using this will have doctests that are ALWAYS skipped. To "break" these users is to actually have their tests run, which doesn't seem too strange.

(This applies equally to any `X.Y` (not just `3.6`), i.e. without the micro version `Z` in `X.Y.Z`.)

The current docs say

> The supported operands are <, <=, ==, >=, >, and comparison is performed by `distutils.version.LooseVersion`.

I'd guess that most Python devs (incl. most that are very experienced) don't know that `LooseVersion` won't compare `3.6` and `3.6.3` as equal. So it would be worth mentioning that. Also, the users don't know **how** "comparison" is performed. They specify **one side**, but the other side is the value of [`platform.python_version()`][1].

But I think just changing the way `compare_version` works is preferable to making the docs longer and harder to read.

[1]: https://github.com/sphinx-doc/sphinx/blob/baaa0e3640eead8e9f926960d990c4e3dea194b8/sphinx/ext/doctest.py#L149

**Comment by shimizukawa** [comment]
How about `~= 3.6`?
https://www.python.org/dev/peps/pep-0440/#version-specifiers

`~=` is supported from setuptools-8.0 that was released at Dec 13, 2014.

**Comment by tk0miya** [comment]
@shimizukawa Good point!
I posted new PR #4393  to follow PEP-440.

@dhermes what do you think about this?

**Comment by tk0miya** [comment]
Now I merged #4393. So please use `~=` operator. It will be released as 1.7.
Thanks,
